flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/DispatcherFailoverITCase.java

```diff
@@ -221,11 +231,11 @@ private TestingDispatcher createRecoveredDispatcher(
             @Nullable FatalErrorHandler fatalErrorHandler) throws Exception {
         final List<JobGraph> jobGraphs = new ArrayList<>();
         for (JobID jobId : haServices.getJobGraphStore().getJobIds()) {
-            jobGraphs.add(haServices.getJobGraphStore().recoverJobGraph(jobId));
+            // there shouldn't be an overlap between dirty JobResults and recovered JobGraphs
+            if (!haServices.getJobResultStore().hasJobResultEntry(jobId)) {
+                jobGraphs.add(haServices.getJobGraphStore().recoverJobGraph(jobId));
+            }
         }
-        // we need to reinstantiate the JobResultStore here to simulate a not-persisting
-        // JobResultStore
-        haServices.setJobResultStore(new EmbeddedJobResultStore());
```

Comment:
FYI: This was the reason the test didn't fail in the previous [FLINK-25430 PR](https://github.com/apache/flink/pull/18189) (`EmbeddedJobResultStore`)